lib/puppet/pops/serialization/to_data_converter.rb

```diff
@@ -92,7 +96,11 @@ def to_data(value)
         elsif @rich_data
           { PCORE_TYPE_KEY => PCORE_TYPE_SYMBOL, PCORE_VALUE_KEY => value.to_s }
         else
-          unknown_to_string_with_warning(value)
+          if @force_symbol
```

Comment:
It feels odd to have both `symbol_as_string` and `force_symbol` as possible parameters. Not to mention that `symbol_as_string` ignores `force_symbol`. Seems that this flag is only applied for the values only, won't we get inconsistencies because the keys are altered?

```ruby
data = {"a" => "b", :c => :d }

Puppet::Pops::Serialization::ToDataConverter.convert(data, symbol_as_string: true, force_symbol: true)
 => {"a"=>"b", "c"=>"d"}

Puppet::Pops::Serialization::ToDataConverter.convert(data, symbol_as_string: false, force_symbol: true)
 => {"a"=>"b", "c"=>:d}
```